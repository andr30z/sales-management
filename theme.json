{
  "color-primary-100": "#FFCFD0",
  "color-primary-200": "#FF9FAA",
  "color-primary-300": "#FF708E",
  "color-primary-400": "#FF4C84",
  "color-primary-500": "#FF1173",
  "color-primary-600": "#DB0C76",
  "color-primary-700": "#B70873",
  "color-primary-800": "#93056A",
  "color-primary-900": "#7A0363",
  "color-success-100": "#D3FDD7",
  "color-success-200": "#A8FCB9",
  "color-success-300": "#7CF6A0",
  "color-success-400": "#5AEE95",
  "color-success-500": "#27e385",
  "color-success-600": "#1CC380",
  "color-success-700": "#13A378",
  "color-success-800": "#0C836B",
  "color-success-900": "#076C62",
  "color-info-100": "#FAFEF3",
  "color-info-200": "#F5FDE8",
  "color-info-300": "#EBF9DA",
  "color-info-400": "#E1F4CE",
  "color-info-500": "#d2edbc",
  "color-info-600": "#A3CB89",
  "color-info-700": "#78AA5E",
  "color-info-800": "#52893B",
  "color-info-900": "#367124",
  "color-warning-100": "#FAD8FF",
  "color-warning-200": "#F2B2FF",
  "color-warning-300": "#E48CFF",
  "color-warning-400": "#D46FFF",
  "color-warning-500": "#bb40ff",
  "color-warning-600": "#932EDB",
  "color-warning-700": "#6E20B7",
  "color-warning-800": "#4E1493",
  "color-warning-900": "#370C7A",
  "color-danger-100": "#FFE6DF",
  "color-danger-200": "#FFC8C0",
  "color-danger-300": "#FFA5A1",
  "color-danger-400": "#FF8A90",
  "color-danger-500": "#FF637A",
  "color-danger-600": "#DB486B",
  "color-danger-700": "#B7315E",
  "color-danger-800": "#931F51",
  "color-danger-900": "#7A1348"
}