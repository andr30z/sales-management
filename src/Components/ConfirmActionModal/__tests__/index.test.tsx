import {
  fireEvent,
  render,
  RenderAPI,
  waitFor,
} from "@testing-library/react-native";
import { ApplicationProvider, Button } from "@ui-kitten/components";
import React, { useState } from "react";
import { ConfirmActionModal, ConfirmActionModalProps } from "..";
import * as eva from "@eva-design/eva";
import { default as theme } from "../../../../theme.json";
describe("<ConfirmActionModal />", () => {
  const openButtonId = "@modal/open-button";
  const TestModal = (
    props: Omit<ConfirmActionModalProps, "open" | "setOpen">
  ) => {
    const [open, setOpen] = useState(false);
    return (
      <ApplicationProvider {...eva} theme={{ ...eva.light, ...theme }}>
        <React.Fragment>
          <ConfirmActionModal open={open} setOpen={setOpen} {...props} />
          <Button testID={openButtonId} onPress={() => setOpen(true)} />
        </React.Fragment>
      </ApplicationProvider>
    );
  };

  const touchables = {
    findOpenButton: (api: RenderAPI) => api.getByTestId(openButtonId),
  };

  it("should render confirmation modal with a different confirm button color", async () => {
    const confirmationText = "ConfirmationText";
    const component = render(
      <TestModal
        onActionConfirmed={() => {
          console.log("I'm confirmed by the user :D!");
        }}
        btnConfirmActionColor="primary"
        confirmText={confirmationText}
      />
    );
    const { getByTestId } = component;
    fireEvent.press(touchables.findOpenButton(component));
    const confirmationButton = await waitFor(() =>
      getByTestId("onPressConfirm")
    );
    const primaryColor = theme["color-primary-500"];
    expect(confirmationButton.props.style.backgroundColor).toBe(primaryColor);
    expect(component.toJSON()).toMatchSnapshot()
  });

  it("should open confirmation modal", async () => {
    const confirmationText = "ConfirmationText";
    const component = render(
      <TestModal
        onActionConfirmed={() => {
          console.log("I'm confirmed by the user :D!");
        }}
        confirmText={confirmationText}
      />
    );
    const { getByText } = component;
    fireEvent.press(touchables.findOpenButton(component));
    const text = await waitFor(() => getByText(confirmationText));
    expect(text).toBeTruthy();
  });

  it("should press confirm button", async () => {
    const onPress = jest.fn();
    const component = render(<TestModal onActionConfirmed={onPress} />);
    const { getByTestId } = component;
    fireEvent.press(touchables.findOpenButton(component));
    await waitFor(() => fireEvent.press(getByTestId("onPressConfirm")));
    expect(onPress).toBeCalled();
  });

  it("should press cancel button and close modal", async () => {
    const confirmMsg = "I'am closed?";
    const component = render(
      <TestModal confirmText={confirmMsg} onActionConfirmed={() => {}} />
    );
    const { getByTestId, queryByText } = component;
    fireEvent.press(touchables.findOpenButton(component));

    const closedTextBeforeModalClose = queryByText(confirmMsg);
    expect(closedTextBeforeModalClose).toBeTruthy();

    await waitFor(() => fireEvent.press(getByTestId("onPressCancel")));
    const closedTextAfterModalClose2 = queryByText(confirmMsg);
    expect(closedTextAfterModalClose2).toBeFalsy();
  });
});
