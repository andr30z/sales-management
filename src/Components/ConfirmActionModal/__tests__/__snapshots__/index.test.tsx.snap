// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmActionModal /> should render confirmation modal with a different confirm button color 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FF1173",
        "borderColor": "#FF1173",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "minHeight": 40,
        "minWidth": 40,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 12,
      }
    }
    testID="@modal/open-button"
  />
  <View
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      backdropStyle={
        {
          "backgroundColor": "rgba(0, 0, 0, 0.5)",
        }
      }
      onBackdropPress={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "height": "32%",
            "width": "95%",
          },
          {
            "position": "absolute",
          },
          {
            "left": -999,
            "top": -999,
          },
        ]
      }
      testID="modalContainer"
      visible={true}
    >
      <View
        style={
          {
            "backgroundColor": "#fff",
            "borderRadius": 13,
            "flex": 1,
            "height": "100%",
            "padding": 10,
          }
        }
      >
        <View
          style={
            {
              "flex": 3,
              "justifyContent": "center",
              "paddingLeft": 5,
            }
          }
        >
          <Text
            appearance="default"
            category="h5"
            style={
              [
                {
                  "color": "#222B45",
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "800",
                },
                {
                  "textAlign": "left",
                },
              ]
            }
            testID="confirmText"
          >
            ConfirmationText
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flex": null,
              "flexDirection": "row",
              "height": "30%",
              "justifyContent": "flex-end",
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#FFFFFF",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 40,
                "minWidth": 40,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 12,
              }
            }
            testID="onPressCancel"
          >
            <Text
              appearance="default"
              category="p1"
              style={
                [
                  {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              Cancelar
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FF1173",
                "borderColor": "#FF1173",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginLeft": 5,
                "minHeight": 40,
                "minWidth": 40,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 12,
              }
            }
            testID="onPressConfirm"
          >
            <Text
              appearance="default"
              category="p1"
              style={
                [
                  {
                    "color": "#222B45",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              Confirmar
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
